packages/mobile/src/send/saga.ts

```diff
@@ -169,12 +184,53 @@ function* sendPaymentOrInviteSaga({
   }
 }
 
+export function* validateRecipientAddressSaga({
+  userInputOfFullAddressOrLastFourDigits,
+  fullAddressValidationRequired,
+  recipient,
+}: ValidateRecipientAddressAction) {
+  Logger.debug(TAG, 'Starting Recipient Address Validation')
+  try {
+    const userAddress = yield select(userAddressSelector)
+    const e164NumberToAddress = yield select(e164NumberToAddressSelector)
+    const { e164PhoneNumber } = recipient
+
+    // should never happen, if it does it means there's an error passing through recipient object
+    if (!e164PhoneNumber) {
+      throw Error('There is no phone number associated with this recipient')
```

Comment:
Can this be checked with types instead

Reply:
Made the error a Recipient one instead so it should be easy to typecheck when we have that option.